paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -335,73 +396,94 @@ def sync_boto_secrets(
                     log.warning(
                         f"Boto key {this_key} required for {service} could not be found."
                     )
+
         if not secret_data:
             continue
+
         # In order to prevent slamming the k8s API, add some artificial delay here
         time.sleep(0.3)
-        app_name = get_kubernetes_app_name(service, instance)
-        namespace = instance_config.get_namespace()
-        secret = limit_size_with_hash(f"paasta-boto-key-{app_name}")
-        hashable_data = "".join([secret_data[key] for key in secret_data])
-        signature = hashlib.sha1(hashable_data.encode("utf-8")).hexdigest()
-        kubernetes_signature = get_kubernetes_secret_signature(
-            kube_client=kube_client,
-            secret=secret,
+        update_k8s_secret(
             service=service,
-            namespace=namespace,
+            secret_name=instance_config.get_boto_secret_name(),
+            secret_data=secret_data,
+            secret_signature=_get_dict_signature(secret_data),
+            kube_client=kube_client,
+            namespace=instance_config.get_namespace(),
         )
-        if not kubernetes_signature:
-            log.info(f"{secret} for {service} in {namespace} not found, creating")
-            try:
-                create_plaintext_dict_secret(
-                    kube_client=kube_client,
-                    secret_name=secret,
-                    secret_data=secret_data,
-                    service=service,
-                    namespace=namespace,
-                )
-            except ApiException as e:
-                if e.status == 409:
-                    log.warning(
-                        f"Secret {secret} for {service} already exists in {namespace} but no signature found. Updating secret and signature."
-                    )
-                    update_plaintext_dict_secret(
-                        kube_client=kube_client,
-                        secret_name=secret,
-                        secret_data=secret_data,
-                        service=service,
-                        namespace=namespace,
-                    )
-                else:
-                    raise
-            create_kubernetes_secret_signature(
+    return True
+
+
+def _get_dict_signature(data: Dict[str, str]) -> str:
+    return hashlib.sha1(
+        "|".join(f"{key}f:{value}" for key, value in data.items()).encode("utf-8")
```

Comment:
```suggestion
        "|".join(f"{key}:{value}" for key, value in data.items()).encode("utf-8")
```

we only need the `f` outside the string :)